ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -217,28 +235,38 @@ static int
 gcp_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session, WT_FILE_SYSTEM *file_system,
   const char *source, const char *object, const char *config)
 {
-    WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
-    WT_UNUSED(config);
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    WT_FILE_SYSTEM *wt_file_system = fs->wt_file_system;
 
-    return 0;
+    // Confirm that the file exists on the native filesystem.
+    std::string src_path = gcp_path(fs->home_dir, source);
+    bool native_exist = false;
+    int ret = wt_file_system->fs_exist(wt_file_system, session, src_path.c_str(), &native_exist);
+    if (ret != 0)
+        return ret;
+    if (!native_exist)
+        return ENOENT;
+    ret = fs->gcp_conn->put_object(object, src_path);
+
+    return ret;
 }
 
 static int
 gcp_flush_finish(WT_STORAGE_SOURCE *storage, WT_SESSION *session, WT_FILE_SYSTEM *file_system,
   const char *source, const char *object, const char *config)
 {
-    WT_UNUSED(storage);
```

Comment:
`storage`, `session`, `config` aren't used in this function yet. Can we keep the WT_UNUSED lines?